File: WebAPI/Models/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace WebAPI.Models
{
    public class RegisterModel
    {
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }

        [DataType(DataType.Password)]
        public string Password { get; set; }
    }
}

File: WebAPI/ClientApp/src/app/models/RegisterModel.ts
export class RegisterModel {
    public name: string;
    public email: string;
    public phone: string;
    public address: string;
    public password: string;

    public constructor(_name: string, _email: string, _phone: string, _address: string, _password: string) {
        this.name = _name;
        this.email = _email;
        this.phone = _phone;
        this.address = _address;
        this.password = _password;
    }
}